Automattic/newspack-plugin
# feat(ras-acc): additional component updates

### All Submissions:


### Changes proposed in this Pull Request:

This PR makes some more updates to the component styles to match the latest mockups. Specifically, it:

#### Adds an example of an error state, and applies those styles to the checkout and sign up modals. 

I approached this like we would add a CSS class around the field that has an error (`.newspack-ui__error`), and then used the existing behaviour (the `400` data-status on the sign-in/up modal and the classes WooCommerce adds) to apply those styles there rather than trying to work in the classes. I'd love to hear if this sounds okay, or terrible!

![image](https://github.com/Automattic/newspack-plugin/assets/177561/e5f6109a-d0a9-4095-8c9e-64863b9bce0d)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/f76154e5-d0e0-4aa0-b0d1-272da92fec7f)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/be01fd46-b24f-403c-a060-7bb14091d37d)

----

#### Reworked the checkbox/radio code to allow for helper text, status labels

Overall, this includes switching the checkbox labels to display as `grid`, and include helper/error text. 

![image](https://github.com/Automattic/newspack-plugin/assets/177561/9441f5c1-5a27-4066-824c-72d8ee1021aa)

I also added an example of a Badge, applied to a input card. This one is set up with the assumption that the badge text will be injected into the `label` when applicable + some `:has` CSS to adjust the grid if the status is there. That might be overkill though -- it can also be reworked to add a CSS class when a label should be applied. This component isn't used in the RAS-ACC styles, but I wanted to make sure it would work with the other checkbox changes:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/927a8077-c553-4561-9e3d-471cc94e4276)

As part of this, I also tweaked the Woo payment option styles to remove some extra space the above added - I basically used the assumption that if there's only one payment option we're hiding the radio button, but just let me know if that's not correct! 

#### Edited to add: updated the newsletter "coda" styles

![image](https://github.com/Automattic/newspack-plugin/assets/177561/fc926c22-1a74-4d43-bf0d-96abf0bfcc28)

----

#### And some small things:
* I set a base font colour and label colour to the styles.
* I renamed the `newspack-ui__input-list` to `newspack-ui__input-card`, which is closer to what it's referred to in Figma.
* I started beefing out the demo content a bit more, adding anchors and some more examples. 
* Reordered the Stripe styles a bit in the Woo overrides SCSS, to keep 'em together.
* Added a style for the 'required' asterisk, following how Woo approaches the markup (`abbr`).

See 1205234045751551-as-1206899964006422

### How to test the changes in this Pull Request:

1. Apply PR and https://github.com/Automattic/newspack-blocks/pull/1719, and run `npm run build`.
2. Turn on the newsletter 'coda' in the checkout by adding `define( 'NEWSPACK_ENABLE_POST_CHECKOUT_NEWSLETTER_SIGNUP', true );` to the wp-config.php file.
3. Log into your test site, and append `?ui-demo` 
4. Review the appearance of the components, and their structure (primarily the changes around checkbox/radio buttons -- does this approach for errors & adding a badge make sense?). 
5. Set up a checkout button block or donate block, and run through the checkout. At each prompt, submit the form without doing anything, or by doing stuff wrong, to see the error styles:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/fcef54b1-fdc0-4c82-a015-021a813ea4cb)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/bf10581d-dfb7-4373-85d4-0aac65fff573)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/4c4f92f9-deac-4e24-a5a0-9084af41822a)

6. Confirm that the newsletter coda is picking up the input 'card' styles -- it's not split into two screens yet (success and newsletter):

![image](https://github.com/Automattic/newspack-plugin/assets/177561/c4cb2f78-d41d-48f4-b5c9-321f01c6dcd1)

### Other information:
